bioptim/examples/getting_started/example_implicit_dynamics.py

```diff
@@ -0,0 +1,197 @@
+"""
+A very simple yet meaningful optimal control program consisting in a pendulum starting downward and ending upward
+while requiring the minimum of generalized forces. The solver is only allowed to move the pendulum sideways.
+
+This simple example is a good place to start investigating explicit and implicit dynamics. There are extra controls in
+implicit dynamics which are joint acceleration qddot thus, u=[tau, qddot]^T. Also a dynamic constraints is enforced at
+each shooting nodes such that InverseDynamics(q,qdot,qddot) - tau = 0.
+
+Finally, once it finished optimizing, it animates the model using the optimal solution.
+"""
+
+import biorbd_casadi as biorbd
+from bioptim import (
+    OptimalControlProgram,
+    DynamicsFcn,
+    Dynamics,
+    Bounds,
+    QAndQDotBounds,
+    InitialGuess,
+    ObjectiveFcn,
+    OdeSolver,
+    CostType,
+    Solver,
+    BoundsList,
+    ObjectiveList,
+)
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+def prepare_ocp(
+    biorbd_model_path: str,
+    final_time: float,
+    n_shooting: int,
+    ode_solver: OdeSolver = OdeSolver.RK1(n_integration_steps=1),
+    use_sx: bool = False,
+    n_threads: int = 1,
+    implicit_dynamics: bool = False,
+) -> OptimalControlProgram:
+    """
+    The initialization of an ocp
+
+    Parameters
+    ----------
+    biorbd_model_path: str
+        The path to the biorbd model
+    final_time: float
+        The time in second required to perform the task
+    n_shooting: int
+        The number of shooting points to define int the direct multiple shooting program
+    ode_solver: OdeSolver = OdeSolver.RK4()
+        Which type of OdeSolver to use
+    use_sx: bool
+        If the SX variable should be used instead of MX (can be extensive on RAM)
+    n_threads: int
+        The number of threads to use in the paralleling (1 = no parallel computing)
+    implicit_dynamics: bool
+        implicit
+    Returns
+    -------
+    The OptimalControlProgram ready to be solved
+    """
+
+    biorbd_model = biorbd.Model(biorbd_model_path)
+
+    objective_functions = ObjectiveList()
+    objective_functions.add(ObjectiveFcn.Lagrange.MINIMIZE_CONTROL, key="tau")
+
+    # Dynamics
+    dynamics = Dynamics(DynamicsFcn.TORQUE_DRIVEN, implicit_dynamics=implicit_dynamics, implicit_soft_contacts=False)
```

Comment:
Yes, I added it because previously we added implicit_soft_contacts as True by default.
I restored implicit_soft_contacts at False by default
and changed the example.

Reply:
That should not be False by default. Default should be True. Why would the default be an unfunctional thing?